src/vmm/src/config/mod.rs

```diff
@@ -158,11 +175,17 @@ impl TryFrom<&str> for KernelConfig {
         // `cmdline=<"string">,path=/path/to/kernel,kernel_load_addr=<u64>`
         // Required: path
         let mut arg_parser = CfgArgParser::new(kernel_cfg_str);
-        let cmdline = arg_parser
+
+        let cmdline_str = arg_parser
             .value_of("cmdline")
             .map_err(ConversionError::new_kernel)?
             .unwrap_or_else(|| DEFAULT_KERNEL_CMDLINE.to_string());
 
+        let mut cmdline = Cmdline::new(KERNEL_CMDLINE_CAPACITY);
+        if cmdline.insert_str(cmdline_str).is_err() {
+            return Err(ConversionError::new_kernel("Kernel cmdline capacity error"));
+        }
```

Comment:
Can we use `map_err` for consistency reasons?

Reply:
sure